hpx/parallel/algorithms/inner_product.hpp

```diff
@@ -72,32 +157,80 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
 
                 difference_type count = std::distance(first1, last1);
 
+                auto f1 =
+                    [op1, op2, policy](
+                        zip_iterator part_begin, std::size_t part_size
+                    ) mutable -> T
+                    {
+                        auto && iters = part_begin.get_iterator_tuple();
+                        FwdIter1 it1 = hpx::util::get<0>(iters);
+                        FwdIter2 it2 = hpx::util::get<1>(iters);
+
+                        FwdIter1 last1 = it1;
+                        std::advance(last1, part_size);
+
+                        if (!util::loop_optimization(policy, it1, last1))
+                        {
+                            // loop_step properly advances the iterators
+                            T result = util::loop_step(parallel::v1::seq,
+                                inner_product_indirect<Op2>{op2}, it1, it2);
+
+                            util::loop2(
+                                parallel::v1::seq, it1, last1, it2,
+                                inner_product_partition<
+                                    parallel::v1::sequential_execution_policy,
+                                    Op1, Op2, T
+                                >{parallel::v1::seq, op1, op2, result});
+
+                            return result;
```

Comment:
The difference between the sequential and the parallel code is that the sequential code works off a initialize value, the parallel code does not (it works on a partition only, and the initial value is applied last).